blackmarket/Networking/Services/AuthenticationServices.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  AuthenticationServices.swift
+//  ios-base
+//
+//  Created by Germán Stábile on 6/8/20.
+//  Copyright © 2020 Rootstrap Inc. All rights reserved.
+//
+
+import RSSwiftNetworking
+
+internal class AuthenticationServices {
+    
+    enum AuthError: Error {
+        case loginError
+        case signUpError
+        case logoutError
+        case mapingError
+        case error
+    }
+    
+    // MARK: - Properties
+    
+    private let sessionManager: SessionManager
+    private let userDataManager: UserDataManager
+    
+    private let apiClient: Client
+    
+    init(
+        sessionManager: SessionManager = .shared,
+        userDataManager: UserDataManager = .shared,
+        apiClient: Client = .shared
+    ) {
+        self.sessionManager = sessionManager
+        self.userDataManager = userDataManager
+        self.apiClient = apiClient
+    }
+    
+    @discardableResult func login(
+        email: String,
+        password: String
+    ) async throws -> Result<UserData, AuthError> {
```

Comment:
If you captured the exception in the function body we can remove throws now. It was the main objective of doing the do catch after all.
```suggestion
    ) async -> Result<UserData, AuthError> {
```

Reply:
Check other functions below as well